docs/develop/best-practices/react.md

```diff
@@ -16,13 +41,176 @@ class MyAccordion extends Component {}
   }
 ```
 
+### On Overridable
+
+See the [How to override UI React components](../howtos/override_components.md) how-to for detailed explanations on the library itself.
+
+The library allows developers to mark components that can be overridden. While it is very easy to use, it can also lead to subtle errors.
+
+#### Overridable components naming
+
+The `id` of each Overridable component should use the following pattern:
+
+```javascript
+<Overridable id="[InvenioModuleName].[ReactApp].[Component].[Element]" name={name}>
+```
+
+- `[InvenioModuleName]`: the Invenio module name, e.g. `InvenioCommunities` or `InvenioAppRDM`.
+- `[ReactApp]`: the name of the web page where the ReactApp will be rendered, e.g. `MembersList`, `MyDashboard`, `MyUploads`, `MyCommunities`, `DepositForm`. There is no exact naming here, try to be consistent with the existing ones.
+- `[Component]`: the name of the component in the React app, e.g. `ResultsList`, `SortBy`, `ListItem`, `BtnAccept`.
+- `[Element]`: the name of the UI section inside the component. `Layout` is normally used of the entire Overridable component, `Title` might be used in an inner section, in case of large components:
```

Comment:
I think it might be slightly misunderstood. I've seen so far something like:
`InvenioAppRDM.ResultList` for entire component (full class replacement), and `InvenioAppRDM.ResultsList.layout` for only JSX overwrite (inside of render method), so maybe we should point out that it is only inside the render method

Reply:
follow up on my previous comment:
we added a type of `.container` to the overridable id, check https://github.com/inveniosoftware/invenio-app-rdm/pull/2087
the change was introduced due to possible confusion which would arise if we call both outer wrapper and inner render of a component with only '.layout' suffix

Reply:
We should have used `CamelCase`, `.Container` :(

Reply:
ouch, we can still change, the PR was not released